auth/src/test/java/org/aerogear/mobile/security/checks/DeveloperModeCheckTest.java

```diff
@@ -0,0 +1,66 @@
+package org.aerogear.mobile.security.checks;
+
+import android.content.Context;
+import android.provider.Settings;
+import android.test.mock.MockContentResolver;
+
+import org.aerogear.mobile.security.SecurityCheckResult;
+import org.aerogear.mobile.security.impl.SecurityCheckResultImpl;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.robolectric.RobolectricTestRunner;
+
+import static junit.framework.Assert.assertEquals;
+import static org.mockito.Mockito.when;
+
+@RunWith(RobolectricTestRunner.class)
+public class DeveloperModeCheckTest {
+    @Mock
+    Context context;
+
+    MockContentResolver mockContentResolver;
+
+    @Before
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void developerModeEnabledTest() {
+        Settings.Global.putInt(mockContentResolver, Settings.Global.DEVELOPMENT_SETTINGS_ENABLED, 1);
+        when(context.getContentResolver()).thenReturn(mockContentResolver);
+
+        SecurityCheckResult expected = new SecurityCheckResultImpl("detectDeveloperMode", true);
+
+        DeveloperModeCheck developerModeCheck = new DeveloperModeCheck();
+
+        SecurityCheckResult actual = developerModeCheck.test(context);
+
+        assertEquals(expected.getName(), actual.getName());
+        assertEquals(expected.passed(), actual.passed());
+    }
+
+    @Test
+    public void developerModeDisabledTest() {
+        Settings.Global.putInt(mockContentResolver, Settings.Global.DEVELOPMENT_SETTINGS_ENABLED, 0);
+        when(context.getContentResolver()).thenReturn(mockContentResolver);
+
+        SecurityCheckResult expected = new SecurityCheckResultImpl("detectDeveloperMode", false);
+
+        DeveloperModeCheck developerModeCheck = new DeveloperModeCheck();
+
+        SecurityCheckResult actual = developerModeCheck.test(context);
+
+        assertEquals(expected.getName(), actual.getName());
+        assertEquals(expected.passed(), actual.passed());
+    }
+
+    @Test(expected = NullPointerException.class)
```

Comment:
@wei-lee just refactored this test.  Not sure if this test adds much value, WDYT?

Reply:
@aidenkeating ^^

Reply:
yeah, I don't think so, but you already have it, so let's keep it. I think it's better to have a sanity check in the implementation to make sure `context` can not be null use the `SanityCheck` class from core.

Reply:
The SantiyCheck (see my comment above), returns an `IllegalArgumentException`, since passing in null, is an illegal argument :-)

Reply:
@rachael-oregan  ^